src/dvc_data/hashfile/meta.py

```diff
@@ -63,12 +63,12 @@ def from_info(
     def from_dict(cls, d: Dict[str, Any]) -> "Meta":
         d = d or {}
         return cls(
-            size=d.pop(cls.PARAM_SIZE, None),
-            nfiles=d.pop(cls.PARAM_NFILES, None),
-            isexec=d.pop(cls.PARAM_ISEXEC, False),
-            version_id=d.pop(cls.PARAM_VERSION_ID, None),
-            etag=d.pop(cls.PARAM_ETAG, None),
-            md5=d.pop(cls.PARAM_MD5, None),
+            size=d.get(cls.PARAM_SIZE),
```

Comment:
How would you know what part is HashInfo (i.e relpath + md5/checksum/etag) vs Meta without popping fields?

Reply:
After spending a bit more time with cloud versioning, HashInfo could be built from a suitable meta field, which is determined by the odb/fs/etc that is being used.